### DISCUSSION TITLE
Static Port Possible on Slingbox?

### DISCUSSION THREAD

**Discussion by donscreen** [GerryDazoo/Slinger#163] [Q&A]
I have a slingbox M2 and slingbox 500.  They both have static IP addresses.
 Occasionally the port number they use changes.  

What triggers a slingbox port change?   Is there somewhere you can specify a static port and have it not change?
Or does the slingbox generate the port number (ie. 5201-5225, or 5301-5325) and increments it when there is some sort of change?   I am trying to understand the port assignment mechanism for slingbox.

When the port number randomly changes...my configs are no longer functional until I put the new port numbers in the configs and update my router port forwarding settings.

**Comment by GerryDazoo** [comment]
I've never had my port number change once it was assigned.

**Comment by DoogieHowitzerWMD** [comment]
Are you setting up the Slinger server in a different location (outside the local network) than where your slingboxes are located?

If so, I experience the same thing, and I have a suspicion as to what is happening.

From my understanding, the slingbox will first try port 5201 and increment from there until it finds a port on which it is allowed to communicate outside of the network. The reason the slingbox is using a different (likely lower) port number than what has been assigned to it is that the security settings on the router are likely set so that devices on your network can negotiate for the connections of the **outgoing** ports.

Right now, I am running the Slinger server in a different location while I test things, and I'm fairly sure that in the home where my slingboxes are physically located, the router its default configuration that allows for devices to attempt to request access to an outgoing port.

I plan on eventually running the Slinger server in the same location as the slingboxes. At that point, the discrepancy with the port assignments shouldn't be an issue, as I believe the router isn't governing any rules with the ports among devices internal to the network, so the server would only be looking to connect to a particular slingbox through a specific port, or not at all.

**Comment by GerryDazoo** [comment]
Just wondering if you're also using the SlingPlayer app or newwatch as well as my code. Maybe it's negotiating a new port to use with the Sling servers.

**Reply by donscreen** [reply]
My slinger server is running on my home office Windows 10 PC on my internal network.  Newwatch no longer works on my mac or windows browsers.  I am running version 3.4 of the Slingplayer Ipad app on my mac some of the time.   I have also used Slingplayer Desktop App on Windows 11 on my mac.
I just now ran the Slingplayer Desktop windows app....and the port changed from 5309 to 5322!

**Reply by DoogieHowitzerWMD** [reply]
On the router where the slingboxes are, I never had to open the ports that the slingboxes currently utilize. My ISP came one day to upgrade to a Gigabit line and put a new router in place. The Slingboxes were already working with the new router before I configured anything, so I didn't bother with the port forwarding that I had done on the previous router. 

When I was planning on doing preliminary setup for the Slinger server, I had to reconfigure the router from a factory reset, as I had somehow locked myself out of accessing the web configuration for it. That is when I opened the ports I planned to use for the each slingbox. However, in the meantime, family were using the slingboxes with Sling's app and servers.

About a week later, when I was getting the Slinger server running on a Raspberry Pi at my house, the only way I was able to get the Slinger server to see my Slingboxes at the other location was to use newwatch on my Windows desktop, go to settings, and check for the port that the Slingbox was utilizing there (ports I never opened in the router). I then had to use those ports in the config file instead of the ones I had originally put there (the ports I was planning to use that I had opened in the router).

Since then, I've only used the Sling App when the Slinger server was not running. However, I can't say for certain about my family staying off the Sling app at the times I've told them to. One of the slingboxes only shows "Weak or No Signal" when connecting to it now, so something happened to it that I'll need to fix at the location.

**Reply by DoogieHowitzerWMD** [reply]
> My slinger server is running on my home office Windows 10 PC on my internal network. Newwatch no longer works on my mac or windows browsers. I am running version 3.4 of the Slingplayer Ipad app on my mac some of the time. I have also used Slingplayer Desktop App on Windows 11 on my mac. I just now ran the Slingplayer Desktop windows app....and the port changed from 5309 to 5322!

That's interesting. I have no idea if my slingboxes will use the ports I specified if run the Sling app on the same internal network. Or maybe I need to soft reset the slingboxes for them to try looking for new ports.

**Reply by donscreen** [reply]
I had uPnP enabled on my router.  That could be the reason the port numbers were changing.   I went in and disabled uPnP on my Asus router.

**Reply by sp222222** [reply]
donscreen, I was about to suggest turning off uPnP. I've never had to change ports on my slingboxes. Although, 3 months ago BEFORE I got my 2nd 500, the original one's port did change because my home routers/switch was replaced (lightning) and even though I assigned it the same IP from static, it did not accept and used another IP address and also another port.
